python_modules/dagster/dagster/_core/storage/alembic/versions/045_1aca709bba64_add_runs_table_backfill_id_index.py

```diff
@@ -0,0 +1,43 @@
+"""add runs table backfill_id index
+
+Revision ID: 1aca709bba64
+Revises: 63d7a8ec641a
+Create Date: 2024-10-23 17:36:39.876876
+
+"""
+
+from alembic import op
+from dagster._core.storage.migration.utils import has_index, has_table
+
+# revision identifiers, used by Alembic.
+revision = "1aca709bba64"
+down_revision = "63d7a8ec641a"
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    if not has_table("runs"):
+        return
+
+    if not has_index("runs", "idx_runs_by_backfill_id"):
+        op.create_index(
+            "idx_runs_by_backfill_id",
+            "runs",
+            ["backfill_id", "id"],
+            unique=False,
+            postgresql_concurrently=True,
```

Comment:
i wonder if you could include `postgresql_where="backfill_id is not null"` and have it operate correctly in sqlite / mysql

Reply:
actually, ignore me... in order to do the exclude_subruns, we want nulls in the index.